### PR TITLE
[pvr] Fix channel scan seg faults, init seg faults and code cleanup

### PR DESCRIPTION
**Pull request by AlwinEsch**
The following changes fix segmention faults during open of PVR channel scan dialog.

### CODE REVIEW

File: xbmc/pvr/PVRManager.cpp [hunk 1]
```diff
@@ -600,7 +603,6 @@ bool CPVRManager::SetWakeupCommand(void)
 
 bool CPVRManager::StartUpdateThreads(void)
 {
-  StopUpdateThreads();
```
Comment (FernetMenta) [inline xbmc/pvr/PVRManager.cpp]: please elaborate on this. why does stopping a thread before starting it fix a segfault? a stopped thread is a pre-condition for starting it.

Comment (AlwinEsch) [inline xbmc/pvr/PVRManager.cpp]: Has added it back, during the last calls it was OK with the rest.

**Comment by AlwinEsch** [comment]
Is something wrong today on github? Also it don't show me there the last update.

**Comment by AlwinEsch** [comment]
OK for you to do a UUID for add-on together with https://github.com/xbmc/xbmc/pull/6877 or on a separate request inside KODI?

**Comment by AlwinEsch** [comment]
Request is fully updated and contains now several changes and fixes.

1.: Fix and possible system freeze on pvr restart due to still required thread work and was stopped.
2.: Fix freeze and seq fault with opened and closed channel scan
3.: Fix the PVR database about faults with already present channels and asked id = -1
4.: Remove a not needed id check, this is always done before
5.: Fix a sometimes produced seq faults during pvr start, which was produced by two threads which has calls sometimes the same EPG database access together.
6.: Remove from pvr system the trailing spaces (don't want to look all the time about it ;-) )
7.: Sort the epg header files alphabetical
8.: Swap over everything there to std::
9.: Use the already defined EPGMAP everywhere
10. Sort also the pvr files alphabetical, also several double headers are removed.

@FernetMenta @opdenkamp can you do a review.
The database part is now really better.

**Comment by FernetMenta** [comment]
looks sane to me.

**Comment by xhaggi** [comment]
@opdenkamp mind taking a look too.

**Comment by fritsch** [comment]
jenkins build this please

**Comment by AlwinEsch** [comment]
Hello @fritsch, there is somewhere a epg database error inside which increase the size of it on every startup, searching currently for it.

**Comment by AlwinEsch** [comment]
Has found the fault, has removed a part which was still required, the related changes was currently not present here but one half of them is the base of this request.

**Comment by AlwinEsch** [comment]
So, everything needed is inside and the bad part is removed.
Allowed to kill me ;-) -->
jenkins build this please

**Comment by fritsch** [comment]
@AlwinEsch Lol - the "shooting" needs ack of the team - and yeah: -1 to that :-)

**Comment by AlwinEsch** [comment]
What the status about this? Merge in?

**Comment by AlwinEsch** [comment]
@opdenkamp has changed to CSingleLock on your node, thanks.

**Comment by fritsch** [comment]
@AlwinEsch this needs a rebase - but we still want it right?

**Comment by AlwinEsch** [comment]
How to handle this at the moment? Ready to merge.
Sorry for the late update, since last days I have not so much time due to private reasons.

**Comment by AlwinEsch** [comment]
With this changes or also without?

**Comment by stefansaraev** [comment]
with this PR, of course. it crashes shortly after I try enter client specific. without this pr - all fine. I can ping you in irc tonight with a propper debug crashlog if you cant reproduce.

**Comment by AlwinEsch** [comment]
Thanks, here it is OK, hope with your crashlog I find the source.

**Comment by stefansaraev** [comment]
ok, it'sa bit more complicated. so:

this PR alone - no crashes. testing with pvr.vdr.vnsi 3a28e39 (without your channelscanner patch). vdr is 2.0.0. vdr-plugin-vnsiserver is 36e2b61 (before your newscanner commits)

this PR + vdr-plugin-vnsiserver-1772c92 (your channelsscaner included) + https://github.com/kodi-pvr/pvr.vdr.vnsi/pull/5 on top of pvr.vdr.vnsi-3a28e39 = crash in tv -> client options, and tv -> general -> search for channels.

call me stupid, but I am lost. not sure if it's this PR or https://github.com/kodi-pvr/pvr.vdr.vnsi/pull/5 is the culprit. 

crashlogs: 
http://sprunge.us/iNML (client options) 
http://sprunge.us/jJDa (channel scanner)

**Comment by MartijnKaijser** [comment]
jenkins build this please
is this something we can shove in 15.0? Seems like a bunch of fixes and needs to go to the beta testers.

**Comment by MartijnKaijser** [comment]
So what's the story n thus PR? Split into smaller parts or this good for merge after another review?

**Comment by opdenkamp** [comment]
fixes should be split up and included in 15, rest should go into a separate pr and pulled later (the cosmetics and increment changes)

**Comment by MartijnKaijser** [comment]
Let me close it for now. New PR can be opened if wanted/needed and is rebased on current master branch.

**Pull request closed without merging**